### DISCUSSION TITLE
Stop specific Cell event propagation when using onRowAction

### DISCUSSION THREAD

**Discussion by lucadido06** [adobe/react-spectrum#6449] [Q&A] [answered]
Hey everyone!

How would you suggest stopping the event propagation from the onRowAction for a specific cell?

Thanks in advance!
Best regards

**Comment by snowystinger** [comment]
Can you say more about your use case? Maybe an example?

**Reply by lucadido06** [reply]
> Can you say more about your use case? Maybe an example?

Hi, thanks for you reply.

Sure, i have a table where clicking on a row redirects to another page but also one of the cell has an action button. When clicking the action button i don't want that the row event is propagated. [Here](https://codesandbox.io/p/sandbox/react-aria-components-stop-propagation-z8tq5r?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clwq5fzqj0006356j1g08f9ub%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clwq5fzqj0002356jj7sec0is%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clwq5fzqj0003356j5nd5jjsy%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clwq5fzqj0005356j1b3pwylo%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clwq5fzqj0002356jj7sec0is%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clwq5fzqj0001356jkngmf90t%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Findex.tsx%2522%257D%255D%252C%2522id%2522%253A%2522clwq5fzqj0002356jj7sec0is%2522%252C%2522activeTabId%2522%253A%2522clwq5fzqj0001356jkngmf90t%2522%257D%252C%2522clwq5fzqj0005356j1b3pwylo%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clwq5fzqj0004356jhsuhq7f4%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clwq5fzqj0005356j1b3pwylo%2522%252C%2522activeTabId%2522%253A%2522clwq5fzqj0004356jhsuhq7f4%2522%257D%252C%2522clwq5fzqj0003356j5nd5jjsy%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clwq5fzqj0003356j5nd5jjsy%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D) you can find an example.

Thanks in advance!

**Reply by snowystinger** [reply, marked as answer]
Sorry, for the delay. Since you're using using your own button, you'll need to stopPropagation on onPointerDown so it doesn't get to the row. `onClick` is problematic for many reasons, they're outlined in https://react-spectrum.adobe.com/blog/building-a-button-part-1.html. It would probably be easier if you could use a RAC Button, or useButton, or usePress (in order of suggested preference). These will all handle stopping the propagation and preventing defaults for you in a way that will most easily work with the rest of our interactions.

**Reply by nicktytarenko** [reply]
thank you @snowystinger it works!
